tests/aws-cpp-sdk-transcribestreaming-integ-tests/TranscribeErrorCaseTests.cpp

```diff
@@ -154,4 +154,179 @@ TEST_F(TranscribeStreamingErrorTests, TranscribeAudioFile) {
+  });
+
+  Aws::String operationRequestId;
+  handler.SetInitialResponseCallback([&](const StartStreamTranscriptionInitialResponse& initialResponse) {
+    operationRequestId = initialResponse.GetRequestId();
+    if (operationRequestId.empty()) {
+      AWS_ADD_FAILURE("InitialResponseCallback is called but received empty RequestId");
+      TestTrace(Aws::String("initialResponse was: ") + initialResponse.Jsonize().View().AsString());
+    }
+    TestTrace(Aws::String("InitialResponse aws RequestId: ") + operationRequestId);
+    TestTrace(Aws::String("InitialResponse transcribe SessionId: ") + initialResponse.GetSessionId());
+  });
+  handler.SetOnErrorCallback([this](const Aws::Client::AWSError<TranscribeStreamingServiceErrors>& errors) {
+    AWS_ADD_FAILURE(Aws::String("Received error: ") + errors.GetExceptionName() + ". with msg: " + errors.GetMessage())
+  });
+
+  StartStreamTranscriptionRequest request;
+  request.SetMediaSampleRateHertz(8000);
+  request.SetLanguageCode(LanguageCode::en_US);
+  request.SetMediaEncoding(MediaEncoding::pcm);
+  request.SetEventStreamHandler(handler);
+
+  auto OnStreamReady = [this](AudioStream& stream) {
+    TestTrace(Aws::String("OnStreamReady"));
+
+    for (size_t i = 0; i < 2; ++i) {
+      Aws::FStream file(TEST_FILE_NAME, std::ios_base::in | std::ios_base::binary);
+      ASSERT_TRUE(file);
+      char buf[2048];
+      while (file) {
+        file.read(buf, sizeof(buf));
+        Aws::Vector<unsigned char> bits{buf, buf + file.gcount()};
+        AudioEvent event(std::move(bits));
+        if (!stream) {
+          break;
+        }
+        TestTrace(Aws::String("Writing good event"));
+        if (!stream.WriteAudioEvent(event)) {
+          AWS_ADD_FAILURE("Failed to write an audio event");
+          break;
+        }
+      }
+      if (i == 0) {
+        TestTrace(Aws::String("Sleeping 10s, shoud get terminated"));
+        std::this_thread::sleep_for(std::chrono::seconds(10));
+      }
+    }
+
+    TestTrace(Aws::String("Sending final empty frame"));
+    stream.WriteAudioEvent({});
+    TestTrace(Aws::String("Flushing and closing the stream"));
+    stream.flush();
+    stream.Close();
+  };
+
+  Aws::Utils::Threading::Semaphore semaphore(0, 1);
+  auto OnResponseCallback = [&semaphore, this](const TranscribeStreamingServiceClient*, const StartStreamTranscriptionRequest&,
+                                         const StartStreamTranscriptionOutcome& outcome,
+                                         const std::shared_ptr<const Aws::Client::AsyncCallerContext>&) {
+    if (outcome.IsSuccess()) {
+      TestTrace(Aws::String("OnResponseCallback, success"));
+    } else {
+      TestTrace(Aws::String("OnResponseCallback, error: ") + outcome.GetError().GetMessage());
+    }
+    semaphore.ReleaseAll();
+  };
+  for (size_t i = 0; i < 2; ++i) {
+    client->StartStreamTranscriptionAsync(request, OnStreamReady, OnResponseCallback, nullptr /*context*/);
+    semaphore.WaitOne();
```

Comment:
please add a reasonable "drop dead" timeout so that we can avoid locking unit tests if something goes wrong